api_server.py

```
@@ -205,6 +204,22 @@ def check_rules(available_ids: List[int]) -> Dict:
+def check_rules(available_ids: List[int]) -> Dict:
+    if not available_ids:
+        return {
+            "rules_pass": False,
+            "rules": {},
+            "skill_counts": {"TTR": 0, "LGV": 0, "BA": 0},
+            "ba_non_ttr": 0,
+        }
+    with get_db() as conn:
+        placeholders = ",".join("?" * len(available_ids))
+        rows = conn.execute(
+            f"SELECT role, skills FROM crew WHERE id IN ({placeholders})", available_ids
```

Ruff (0.15.4)

[error] Possible SQL injection vector through string-based query construction

(S608)